cpp/src/distance/detail/fused_distance_nn.cuh

```diff
@@ -68,16 +69,31 @@ void fusedDistanceNNImpl(OutT* min,
 
   switch (metric) {
     case cuvs::distance::DistanceType::CosineExpanded:
-      fusedCosineNN<DataT, OutT, IdxT, P, ReduceOpT, KVPReduceOpT>(
-        min, x, y, xn, yn, m, n, k, workspace, redOp, pairRedOp, sqrt, stream);
+      if constexpr (std::is_same_v<DataT, uint8_t> || std::is_same_v<DataT, int8_t>) {
+        assert(false && "Cosine distance is not supported for uint8_t/int8_t data types");
```

Comment:
These `asserts` will only trigger if the code is compiled in debug mode. I don't think this is what we want.